File: packages/app/src/services/software/definitions/arc.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'arc';
	public name = 'Arc';
	public category = [SoftwareCategory.Browser];
	public downloadName = 'ArcInstaller.exe';
	public icon = 'arc.png';
	public homepage = 'https://arc.net';

	public async resolveDownloadUrl() {
		return ok('https://releases.arc.net/windows/ArcInstaller.exe');
	}
}

File: src/lib/software/arc.py
from src.lib.software import BaseSoftware, SoftwareCategory

class Arc(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'arc'
        self.name = 'Arc'
        self.category = [SoftwareCategory.Browser]
        self.download_name = 'ArcInstaller.exe'
        self.icon = 'arc.png'
        self.homepage = 'https://arc.net'

    def resolve_download_url(self):
        self.url_resolved.emit('https://releases.arc.net/windows/ArcInstaller.exe')
